### PR TITLE
box: remove `cord_slab_cache` symbol from public API export

### LINKED ISSUE #7124: cord_slab_cache(): Assertion `pthread_equal(cache->thread_id, pthread_self())' failed.
**Bug description**

Assertion fails when cord_slab_cache (got from Tarantool's C API) is being used as a backend for a libsmall (master version) region allocator in the latest release candidate Tarantool.

* OS: Linux 5.4
* OS Version: CentOS 8.5
* Architecture: x86_64

```
tarantool --version
Tarantool 2.10.0-rc1-0-g7ed15e62b
Target: Linux-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_BACKTRACE=ON
Compiler: /usr/bin/cc /usr/bin/c++
C_FLAGS:-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches  -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -std=c11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type -Werror
CXX_FLAGS:-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches  -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -std=c++11 -Wall -Wextra -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror
```

**Steps to reproduce**

*csc_issue.c*
```
#include <string.h>

#include <tarantool/lua.h>
#include <tarantool/lauxlib.h>
#include <tarantool/module.h>

#include <small/region.h>

#define TEST_DATA "Hello world"

static int
csc_issue_use_csc_from_c(lua_State *L)
{
        struct region gc;
        region_create(&gc, cord_slab_cache());

        char *test_data = (char *)region_alloc(&gc, sizeof(TEST_DATA));
        if (!test_data)
                return luaL_error(L, "ENOMEM");

        memcpy(test_data, TEST_DATA, sizeof(TEST_DATA));
        say_info("%.*s", (int)sizeof(TEST_DATA), test_data);

        region_destroy(&gc);

        return 0;
}

static const struct luaL_Reg
R[] = {
        {"use_csc_from_c", csc_issue_use_csc_from_c},
        {NULL, NULL},
};

LUA_API int
luaopen_csc_issue(lua_State *L)
{
        luaL_newmetatable(L, "csc_issue");
        lua_pushvalue(L, -1);
        lua_setfield(L, -2, "__index");
        luaL_register(L, NULL, R);
        return 1;
}
```

*reproducer.lua*
```
local csc_issue = require('csc_issue')
csc_issue.use_csc_from_c()
```

How to run:

```
git clone git@github.com:tarantool/small.git
cmake3 -DSMALL_EMBEDDED=1 -DCMAKE_C_FLAGS="-std=c99 -fPIC" -S small -B small && cmake3 --build small
gcc -shared -Wall -fPIC -std=c99 -I small/include/ -I small/third_party/ -o csc_issue.so csc_issue.c -Lsmall -lsmall
tarantool reproducer.lua
```

**Actual behavior**

Call to `region_alloc` causes assertion fail.

**Expected behavior**

Successful memory allocation.

### PR DESCRIPTION
**Pull request by CuriousGeorgiy**
Closes #7124

### CODE REVIEW

File: src/lib/core/fiber.h [hunk 1]
```diff
@@ -172,6 +172,12 @@ enum {
 	FIBER_DEFAULT_FLAGS	= 0
 };
 
+/**
+ * Return slab_cache suitable to use with tarantool/small library
+ */
+struct slab_cache *
+cord_slab_cache(void);
+
```
Comment (locker) [inline src/lib/core/fiber.h:180]: Nit: Let's make it inline as it's a one-liner.

File: test/app-tap/module_api.c [hunk 2]
```diff
@@ -220,6 +228,8 @@ test_box_ibuf(lua_State *L)
 	fail_unless(ibuf_used(&ibuf) == 0);
 	fail_unless(*rpos == *wpos);
 
+	slab_arena_destroy(&arena);
+
```
Comment (locker) [inline test/app-tap/module_api.c:232]: Nit: `slab_cache_destroy` missing.

Comment (CuriousGeorgiy) [reply test/app-tap/module_api.c:232]: Doesn't the arena own the slab cache's memory?

**Review by andreyaksenov — APPROVED** [review]

**Comment by locker** [comment]
ASAN workflow fails. Please take a look.

**Comment by CuriousGeorgiy** [comment]
@locker we discussed this problem with @Totktonada, and I removed the dependency from small by pushing an `ibuf` from Lua.

**Review by locker — APPROVED** [review]

**Pull request merged**